### DISCUSSION TITLE
Can't edit or create items when accessing through the 'Organizations' tab

### DISCUSSION THREAD

**Discussion by Alintya** [dani-garcia/vaultwarden#3371] [Q&A] [answered]
### Subject of the issue
When accessing the vault through the Organizations tab, the edit item modal is empty and trying to create a new item throws an error on save, seemingly returning an entire html page.

### Your environment (Generated via diagnostics page)

* Vaultwarden version: v1.27.0
* Web-vault version: v2022.12.0
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: false (actually true)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: false
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.39.2
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)

<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** SIGNUPS_ALLOWED, INVITATIONS_ALLOWED, SHOW_PASSWORD_HINT, ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*********************",
  "domain_origin": "*****://*********************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "~",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "Login",
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "***********************",
  "smtp_from_name": "~",
  "smtp_host": "*************",
  "smtp_password": "***",
  "smtp_port": 465,
  "smtp_security": "force_tls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "***********************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

* Install method: Docker on Unraid OS

* Clients used: web vault, (windows Bitwarden app)

* Reverse proxy and version: nginx via swag docker container (2.4.0)

### Steps to reproduce

- (own an organization)
- login to web vault
- go to Organizations tab
- click any item or try to add one

### Expected behaviour

Editing or editing items works as is the case on the 'Vaults' tab

### Actual behaviour

- Editing shows an empty modal (except for the headline)
- Adding throws an error on save

### Troubleshooting data

![image](https://user-images.githubusercontent.com/12158187/227781565-9c8dc71a-6f6b-4b5a-9b5c-9c49e817d7e8.png)

![image](https://user-images.githubusercontent.com/12158187/227781835-1c4fa851-f874-4e41-9ec7-66c93ca13dc6.png)

**Comment by BlackDex** [comment]
What do the logs say of both Vaultwarden and the reverse proxy.
Does it also use ModSecurity for example? That tends to break stuff, like PUT and DELETE requests.

It works fine for me btw, no issues at all.

**Comment by Alintya** [comment]
Are there additional log files? the console doesn't produce any records when clicking edit or save
```
[2023-03-26 16:37:17.284][start][INFO] Rocket has launched from http://0.0.0.0:80
[2023-03-26 16:38:09.131][request][INFO] POST /identity/connect/token
[2023-03-26 16:38:09.158][response][INFO] (login) POST /identity/connect/token => 200 OK
[2023-03-26 16:38:09.206][request][INFO] GET /api/sync?excludeDomains=true
[2023-03-26 16:38:09.226][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
[2023-03-26 16:38:15.789][request][INFO] GET /api/organizations/73f0daab-2bc7-4d7b-826e-82721d4d17c6/collections
[2023-03-26 16:38:15.790][response][INFO] (get_org_collections) GET /api/organizations/<org_id>/collections => 200 OK
[2023-03-26 16:38:15.792][request][INFO] GET /api/organizations/73f0daab-2bc7-4d7b-826e-82721d4d17c6/collections
[2023-03-26 16:38:15.792][response][INFO] (get_org_collections) GET /api/organizations/<org_id>/collections => 200 OK
[2023-03-26 16:38:15.871][request][INFO] GET /api/ciphers/organization-details?organizationId=73f0daab-2bc7-4
[2023-03-26 16:38:15.878][response][INFO] (get_org_details) GET /api/ciphers/organization-details?<data..> => 200 OK
```

**Comment by BlackDex** [comment]
If that is the case then something is blocked by your reverse proxy.

**Comment by Alintya** [comment]
I chose not to forward `/admin` but other than that i stuck to the example shipped with swag found [here](https://github.com/linuxserver/reverse-proxy-confs/blob/master/vaultwarden.subdomain.conf.sample)
Can you spot anything wrong with that?

**Comment by Alintya** [comment]
Well, it turns out that is exactly the problem; also forwarding `/admin` resolves the issue.
This shouldn't be the case though I guess?

I would want to restrict access to the admin panel on a (semi-)public instance but that will apparently also apply to people using the organizations tab.
I just tried putting http auth on the `/admin` route for example, which sure enough got triggered by clicking an item to edit

**Reply by BlackDex** [reply]
You should match any url **starting** with `/admin` not when it is anywhere else in the url like `/cipers/<uuid>/admin` because that is probably triggered now also, and that is wrong.

**Comment by stefan0xC** [comment, marked as answer]
The location matching is not specific enough by using `~` you are using regular expression. You want to restrict paths starting with `/admin` and not everything containing `/admin`.
